storagemarket/deal_handler_test.go

```diff
@@ -0,0 +1,64 @@
+package storagemarket
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCancellationSimple(t *testing.T) {
+	_, tCancel := context.WithCancel(context.Background())
+
+	dh := &dealHandler{
+		providerCtx:    context.Background(),
+		transferCancel: tCancel,
+		transferDone:   make(chan error, 1),
+	}
+
+	require.False(t, dh.TransferCancelledByUser())
+	dh.transferCancelled(nil)
+	require.NoError(t, dh.cancel())
+	require.True(t, dh.TransferCancelledByUser())
+}
+
+func TestCancellationByCancel(t *testing.T) {
```

Comment:
Same comment here - can we add godoc on what this method is testing?

Reply:
done.